### PR TITLE
BOLT catchup, including dual funding!

### PR DESCRIPTION
**Pull request by rustyrussell**
@niftynei  I can't find one quote: does the 110 minimum witness weight thing still apply?  See the second-last commit...

### CODE REVIEW

**Review by vincenzopalazzo — COMMENTED** [review]
CI catch something that I was able to reproduce also on my host

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================================================= short test summary info ========================================================================================================================
FAILED tests/test_opening.py::test_rbf_non_last_mined - AssertionError: assert 'f8ff15a3f0d2...87a6a12f4bfe8' == 'f5fecb7b2b00...bd148b580a23e'
```

**Comment by niftynei** [comment]
>  does the 110 minimum witness weight thing still apply?

No it was removed in favor of no restrictions on witness weight payments!

**Comment by rustyrussell** [comment]
> > does the 110 minimum witness weight thing still apply?
> 
> No it was removed in favor of no restrictions on witness weight payments!

Does code need to change, or just comments?

**Comment by endothermicdev** [comment]
I tried doing the obvious thing, removing the minimum witness weight requirement as it was removed from the spec.  Correct me if I'm wrong @niftynei!

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -924,12 +924,12 @@ static void NON_NULL_ARGS(1, 2, 4, 5) json_add_channel(struct lightningd *ld,
...
-		 * - MUST set `funding_feerate_perkw` greater than or equal to
-		 *   65/64 times the last sent `funding_feerate_perkw`
-		 *   rounded down.
+		/* BOLT #2:
+		 * - MUST set `feerate` greater than or equal to 25/24
+		 * times the `feerate` of the previously constructed
+		 * transaction, rounded down.
 		 */
-		next_feerate = last_feerate * 65 / 64;
+		next_feerate = last_feerate * 25 / 24;
```
Comment (endothermicdev) [inline lightningd/peer_control.c:932]: Still kind of weird to round down a `>=`, but it's nice that the spec is explicit.

File: openingd/dualopend.c [hunk 2]
```diff
@@ -3359,15 +3359,14 @@ static bool check_funding_feerate(u32 proposed_next_feerate,
...
-	 * - MUST fail the negotiation if:
-	 *   - the `funding_feerate_perkw` is not greater than 65/64 times
-	 *   `funding_feerate_perkw` of the last successfully negotiated
-	 *   open attempt
+	 *   - MUST respond with `tx_abort` if:
+	 *     - the `feerate` is not greater than or equal to 25/24 times `feerate`
+	 *       of the last successfully constructed transaction
 	 */
-	u32 next_min = last_feerate * 65 / 64;
+	u32 next_min = last_feerate * 25 / 24;
```
Comment (endothermicdev) [inline openingd/dualopend.c:3394]: And here it bites us.  The spec should be updated with the "round down" verbiage here too.

**Review by endothermicdev — APPROVED** [review]
Okay, I see how directly quoting the spec is basically a superpower.
ACK b60c512a6edaeac42266b33f8ac4311f225a61a7

**Comment by endothermicdev** [comment]
Rebased on master with renepay refactor.

**Pull request merged**